lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -54,19 +41,28 @@ def timeout
       self.class.timeout
     end
 
-    def initialize(name:, namespace:, context:, file:, logger:)
+    def initialize(namespace:, context:, template:, logger:)
       # subclasses must also set these if they define their own initializer
-      @name = name
+      @name = template.fetch("metadata", {})["name"]
       @namespace = namespace
       @context = context
-      @file = file
       @logger = logger
+      @template = template
```

Comment:
True, it's the entire resource definition, which actually contains things called "template". I'm hesitant to call in JSON since it is parsed. Kubernetes docs refer to this as [Kubernetes Objects](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/), but I don't think that would really be clearer (especially given the significance of "object" in Ruby). How do you feel about `resource_definition`?